cli/azd/resources/starter/.github/workflows/azd-up.yml

```diff
@@ -0,0 +1,73 @@
+name: Deploy up to Azure
+
+on:
+  workflow_dispatch:
+  push:
+    # Run when commits are pushed to mainline branch (main or master)
+    # Set this to the mainline branch you use
+    branches:
+      - main
+      - master
+
+# Example GitHub Actions workflow to deploy to Azure using azd
+# To configure required secrets for connecting to Azure, simply run `azd pipeline config`
+
+# Set up permissions for deploying with secretless Azure federated credentials
+# https://learn.microsoft.com/en-us/azure/developer/github/connect-from-azure?tabs=azure-portal%2Clinux#set-up-azure-login-with-openid-connect-authentication
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container:
+      # Use azd provided container image that has azd, az, multi-language build tools pre-installed.
+      image: mcr.microsoft.com/azure-dev-cli-apps:latest
+    env:
+      AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+      AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+      AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Log in with Azure (Federated Credentials)
+        if: ${{ env.AZURE_CLIENT_ID != '' }}
+        run: |
+          azd auth login `
+            --client-id "$Env:AZURE_CLIENT_ID" `
+            --federated-credential-provider "github" `
+            --tenant-id "$Env:AZURE_TENANT_ID"
+        shell: pwsh
+
+      # Remove if you don't need to use client credentials
+      # By default, `azd pipeline config` will configure federated credentials.
+      - name: Log in with Azure (Client Credentials)
+        if: ${{ env.AZURE_CREDENTIALS != '' }}
+        run: |
+          $info = $Env:AZURE_CREDENTIALS | ConvertFrom-Json -AsHashtable;
+          Write-Host "::add-mask::$($info.clientSecret)"
+
+          azd auth login `
+            --client-id "$($info.clientId)" `
+            --client-secret "$($info.clientSecret)" `
+            --tenant-id "$($info.tenantId)"
+        shell: pwsh
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+
+      - name: Provision Infrastructure
```

Comment:
Perhaps include a comment in here as well to say `azd provision` is optional if resources have been provisioned and do not require any update. Customers ca have 2 pipelines - one for infra; another one for app.

Reply:
IMO, addressing the issue in question means scaffolding the absolute MVP of a skeleton. In other words, spitting out a skeleton that meets azd conventions to be recognized as template (e.g. infra/ and azure.yaml). Everything else is bonus. 

I think we can avoid scope creeping this issue and do further research on the various levels of scaffolding that folks might want as part of the rapid prototyping work we have planned for this semester.

Reply:
I agree; no scope creep. My suggestion is to include a comment to point out customers can omit the `azd provision` step if they desire.

Reply:
> @puicchan Do you see this being widely used in practice? I grew up in services that always recommended running both IaC provisioning and code deployment regularly. And infra and code certainly can be separated out (and we did for some things), but I'm not sure what our default starter should do. I think the smart default here is to combine the two.
> 
> But maybe we need different scenario example files? @savannahostrowski 😄 This may be the spiral I was referring to.

Yes, used to own Jenkins on Azure plugin but we don't need to address those scenarios. Just one simple sample will do.